src/execution_tree/primitives/column_slicing.cpp

```diff
@@ -51,83 +52,97 @@ namespace phylanx { namespace execution_tree { namespace primitives
+            if (start < 0)
             {
-                // return the input as it is if the input is of zero dimension or
-                // one dimension. The values passed to col_start, col_stop
-                // does not have an effect on the result.
+                actual_start = array_length + start;
+            }
 
-                return primitive_argument_type(std::move(args[0]));
+            if (stop >= 0)
+            {
+                actual_stop = stop;
+            }
+
+            if (stop < 0)
+            {
+                actual_stop = array_length + stop;
             }
 
-            primitive_argument_type column_slicing1d(args_type && args) const
+            std::vector<int> result;
+
+            if (step > 0)
             {
-                // return elements starting from col_start to col_stop(exclusive)
-
-                auto col_start = args[1][0];
-                auto col_stop = args[2][0];
-
-                // parameters required by blaze to create a submatrix is as follows:
-                // subvector(vector,column,n)
-                // vector The vector containing the subvector.
-                // column The index of the first column of the subvector.
-                // n The number of columns of the subvector.
-                // return View on the specific subvector of the vector.
-
-                // The following math is a result of converting the arguments
-                // provided in slice primitive so that equivalent operation is
-                // performed in blaze.
-                // vector = vector
-                // column = col_start
-                // n = (col_stop - col_start)
-
-                if (col_start < 0 && col_stop > 0)    // slice from the end
+                for (int i = actual_start; i < actual_stop; i += step)
                 {
-                    HPX_THROW_EXCEPTION(hpx::bad_parameter,
-                        "phylanx::execution_tree::primitives::"
-                            "column_slicing_operation::column_slicing_operation",
-                        "col_stop can not be positive if col_start is negative");
+                    result.push_back(i);
                 }
-                if (col_start >= 0 && col_stop < 0)
+            }
+
+            if (step < 0)
+            {
+                for (int i = actual_start; i > actual_stop; i += step)
                 {
-                    HPX_THROW_EXCEPTION(hpx::bad_parameter,
-                        "phylanx::execution_tree::primitives::"
-                            "column_slicing_operation::column_slicing_operation",
-                        "col_stop can not be negative if col_start is positive");
+                    result.push_back(i);
                 }
+            }
 
-                using storage0d_type = typename arg_type::storage0d_type;
-                using storage1d_type = typename arg_type::storage1d_type;
+            //Note: std::vector will throw Invalid element access index if
```

Comment:
I'd suggest to explicitly throw an exception here with an appropriate error message instead of relying for the vector to diagnose out-of-bounds indexing.